lib/memjs/memjs.js

```diff
@@ -170,6 +186,137 @@ Client.prototype.get = function(key, callback) {
   });
 };
 
+/** Build a pipelined get multi request by sending one GETKQ for each key (quiet, meaning it won't respond if the value is missing) followed by a no-op to force a response (and to give us a sentinel response that the pipeline is done)
+ *
+ * cf https://github.com/couchbase/memcached/blob/master/docs/BinaryProtocol.md#0x0d-getkq-get-with-key-quietly
+ */
+Client.prototype._buildGetMultiRequest = function(keys) {
+  // start at 24 for the no-op command at the end
+  var requestSize = 24;
+  for (var keyIdx in keys) {
+    requestSize += Buffer.byteLength(keys[keyIdx], 'utf8') + 24;
+  }
+
+  var request = Buffer.alloc(requestSize);
+  request.fill();
+
+  this.incrSeq();
+
+  var bytesWritten = 0;
+  for (keyIdx in keys) {
+    var key = keys[keyIdx];
+    bytesWritten += addToRequestBuffer(constants.OP_GETKQ, key, '', '', this.seq, request, bytesWritten);
+  }
+
+  bytesWritten += addToRequestBuffer(
+    constants.OP_NO_OP, '', '', '', this.seq, request, bytesWritten);
+
+  return request;
+};
+
+/** Executing a pipelined (multi) get against a single server. This is a private implementation detail of getMulti. */
+Client.prototype._getMultiToServer = function (serv, keys, callback) {
+  var self = this;
+
+  var responseMap = {};
+
+  var handle = function(response) {
+    switch (response.header.status) {
+    case  0:
+      if (callback) {
+        var deserialized = self.serializer.deserialize(response.header.opcode, response.val, response.extras);
+        // When we get the no-op response, we are done with this one getMulti in the per-backend fan-out
+        if (response.header.opcode === constants.OP_NO_OP) {
+          // this ensures the handler will be deleted from the responseCallbacks map in server.js
+          handle.quiet = true;
```

Comment:
This should be `handle.quiet = false`

Reply:
Although maybe it doesn't matter if the no-op body length is zero, because then it would be deleted anyways here:
https://github.com/makenotion/memjs/blob/021cbbe04c3aa6e6929e64642d30f451d4518322/lib/memjs/server.js#L46

still i think you intended `handle.quiet = false`

Reply:
I did, added a comment also.

Also added a bunch of tests for cleaning up callbacks which I'm happy to say caught some bugs